src/app/context/Bib.Provider.jsx

```diff
@@ -78,10 +78,10 @@ function extractParallels(bib) {
         // If parallel but no none parallel (original) match
         if (!bib[field]) return store;
 
-        const mapping = bib[key]
+        const mapping = bib[field]
           .reduce((acc, curr, idx) => {
-            const pa = curr;
-            const og = bib[field][idx];
+            const og = curr;
+            const pa = bib[key][idx];
             // @seanredmond would like the parallel to show up first
             // https://jira.nypl.org/browse/SCC-2915?focusedCommentId=68727&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-68727
             const ne = [pa, og];
```

Comment:
You're building a flattened list of values to display. The order of the values is correct (parallel values always immediately precede the "original" value). But because it's a flattened array, you lose the formal relationship between original values and their parallels. The relationship is implicit in the order. This means, when displaying the list of values (parallels interleaved in originals), you don't have a clean way of identifying which value is the original and which is the parallel. In the case of Notes you can [rely on the quirk](https://github.com/NYPL/discovery-front-end/pull/1887/files#diff-1d43beab747863937d2170f28c9824653d2983ca643e535028e4dc1cf31610b3R21-R24) that the parallel value for a note is always a String (instead of an object). But relying on convention to distinguish original values from parallels feels brittle. And I do suspect the designs will eventually require us to differentiate original values from parallel values (e.g. we may need to force a different font on parallel values, or we may want to adjust padding to clearly indicate a relationship between a parallel value and the original value it refers to)

What if, when doing your preprocessing of original values (looking for parallels), rather than produce a combined, flattened array, simply attach the parallel value to the original note object. Before grouping notes on their types (because you lose the original indexes after that) loop over the original notes and simply attach the parallel like `notes[idx].parallel = parallelNotes[idx]`. That way, when rendering the notes, you don't need to [use a different list of notes depending on whether or not there are any parallels defined](https://github.com/NYPL/discovery-front-end/pull/1887/files#diff-1d43beab747863937d2170f28c9824653d2983ca643e535028e4dc1cf31610b3R6-R11). You can just loop over the list of notes and whether/not the earlier process attached parallels to some/all/none of the values, you just need to check for `note.parallel` [here](https://github.com/NYPL/discovery-front-end/pull/1887/files#diff-1d43beab747863937d2170f28c9824653d2983ca643e535028e4dc1cf31610b3R21-R24) to determine whether or not to display the extra `<li/>` with the parallel. 

I recognize that approach depends on the "original" values already being objects. If you took the same approach with other original values (e.g. contributorLiteral), you couldn't attach a `.parallel` property to the original string. In that case, your preprocessing step might opt to convert the original string values into objects, structured like `{ original: "og value", parallel "parallel value" }`. When rendering the value in the `<dl/>` you'd need to check for the `typeof` the value; If it's a string, there's no parallel. If it's an object with a `.parallel`, you know you need to first render the parallel.

Alternatively, don't preprocess the list of values at all. I don't see a reason you strictly need to preprocess the list of values to display. You could retrieve the parallel at render time 
```
              getParallelValue(bib, field, idx) {
                const parallelField = `parallel${capitalize(field)}`
                if (!bib[parallelField]) return null
                return bib[parallelField][idx]  
              }
              
              ...
                {(getParallelValue(bib, 'contributorLiteral', idx) && (
                  <li key={`${idx}-parallel`} className="special-parallel-class">{getParallelValue("contributorLiteral", idx)}</li>
                )) ||
                  null}
```

Essentially, at render time, assuming you have the bib, the field name, and the original index, you can get the parallel on the fly. (This won't work for notes because the front-end is grouping them by type, but there are ways to preserve that original index - either by storing the original indexes somewhere before grouping, or attaching the original indexes to the notes before they're re-grouped..)